src/Donations/Repositories/DonationRepository.php

```diff
@@ -468,6 +468,27 @@ public function getFormTitle($formId)
         return $form->post_title;
     }
 
+    /**
+     * @unreleased
+     *
+     * @param int $donationId
+     *
+     * @return string
+     */
+    public function getFormTitleByDonationId($donationId)
+    {
+        $metaData = DB::table('give_donationmeta')
+            ->where('donation_id', $donationId)
+            ->where('meta_key', '_give_payment_form_title')
+            ->get();
+
+        if (!$metaData) {
+            return '';
+        }
+
+        return $metaData->meta_value;
+    }
```

Comment:
@ravinderk Why not use the `$donation->formTitle` model property?

Reply:
@JasonTheAdams So you are recommending the creation of a Donation model to access the form title instead of using the repository.

Reply:
Yes. Think of the models as the first-class way of getting donation data. We want to minimize specific queries like this and simply do things like:
```php
$donation = Donation::find($donationId); // null if not found
$donation->formTitle;
```

Ultimately, we can add caching around retrieving models. In any case, this should be the preferred means of retrieving individual Donation data.

Reply:
@JasonTheAdams Yes, I was thinking about caching. I will implement it without any issue

Reply:
To be clear, we won't worry about adding caching now. It's something we'll tackle later. Honestly, just reducing the number of random meta queries will alone improve performance.